File: day02/day02.py

with open ("input.txt", "r") as f:
    data = f.read().splitlines()


# part 1
player = {
    "X" : 1,
    "Y" : 2,
    "Z" : 3,
}

opponent = {
    "A" : 3,
    "B" : 2,
    "C" : 1,
}

points = 0

# part 2
opponent_2 = {
    "A" : 1,
    "B" : 2,
    "C" : 3,
}
points_2 = 0

for l in data:
    
    op, pl = l.split(" ")

    points += ( player[pl] * 3 + opponent[op] * 3)%9 + player[pl]

    points_2 += (player[pl] - 1 ) * 3 + (player[pl] + opponent_2[op]) % 3 + 1 ; 

print("challenge 1: ", points)
print("challenge 2: ", points_2)

File: day02/day02.cpp
#include <iostream>
#include <fstream>
#include <sstream>
#include <unordered_map> 

using std::istringstream;
using std::ifstream;
using std::string;
using std::cout;
using std::unordered_map;

int main(){


    string input;

    char opponent;
    char player;

    // part 1
    unordered_map<char, int> playerHands = {
        {'X', 1},
        {'Y', 2},
        {'Z', 3}
    };

    unordered_map<char, int> opponentHands_1 = {
        {'A', 3},
        {'B', 2},
        {'C', 1}
    };

    int points_1 = 0;
    
    // part 2
    unordered_map<char, int> opponentHands_2 = {
        {'A', 1},
        {'B', 2},
        {'C', 3}
    };
    
    int points_2 = 0; 

    ifstream inputFile("input.txt");
    while (getline(inputFile, input)){

        istringstream parsingStream(input);
        
        parsingStream >> opponent;
        parsingStream >> player;

        points_1 += (playerHands[player] * 3 + opponentHands_1[opponent] * 3) % 9 + playerHands[player]; 

        points_2 += (playerHands[player] - 1 ) * 3 + (playerHands[player] + opponentHands_2[opponent]) % 3 + 1 ; 

    } 

    
    inputFile.close();


    printf("Part 1: %i\n", points_1);
    printf("Part 2: %i\n", points_2);

    return 1;
}